### PR TITLE
feat: [STO-5355]: Add ticket account settings page

### PR DESCRIPTION
**Pull request by ahokinson**
#### Screenshots

CleanShot 2023-01-22 at 23 54 23@2x

#### PR Checklist

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/10-common/RouteDefinitions.ts [hunk 1]
```diff
@@ -1903,6 +1903,8 @@ const routes = {
       `/sto/orgs/${orgIdentifier}/projects/${projectIdentifier}/getting-started`
   ),
   /********************************************************************************************************************/
+  toTickets: withAccountId(() => `/settings/tickets`),
```
Comment (chris-harness) [inline src/modules/10-common/RouteDefinitions.ts:1906]: Move this up to line 153 or so with the other settings routes

File: src/modules/10-common/components/AccountSideNav/AccountSideNav.tsx [hunk 2]
```diff
@@ -51,6 +57,9 @@ export default function AccountSideNav(): React.ReactElement {
       )}
       <SidebarLink label={getString('common.auditTrail')} to={routes.toAuditTrail({ accountId })} />
       <SidebarLink label={getString('orgsText')} to={routes.toOrganizations({ accountId })} />
+      {STO_JIRA_INTEGRATION && (
+        <SidebarLink label={getString('common.jira.tickets')} to={routes.toTickets({ accountId })} />
```
Comment (chris-harness) [inline src/modules/10-common/components/AccountSideNav/AccountSideNav.tsx:61]: This isn't going to be Jira specific in the future, so omit `jira`

File: src/modules/70-pipeline/strings/strings.en.yaml [hunk 3]
```diff
@@ -148,8 +148,8 @@ jenkinsStep:
   captureEnvironmentVariable: Capture environment variables from Jenkins build
 jiraApprovalStep:
   connectToJira: Connect to Jira
-  connectorRef: Jira Connector
-  jiraConnectorPlaceholder: Select Jira Connector
+  connectorRef: '{{$.common.jira.connector}}'
+  jiraConnectorPlaceholder: '{{$.common.jira.selectConnector}}'
```
Comment (chris-harness) [inline src/modules/70-pipeline/strings/strings.en.yaml:152]: Revert these and change our strings to "[Select] Ticket Provider Connector"

File: src/modules/10-common/pages/Tickets/Tickets.tsx [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+  const { accountId, projectIdentifier, orgIdentifier } =
+    useParams<PipelineType<PipelinePathProps & AccountPathProps>>()
+  const { repoIdentifier, branch } = useQueryParams<GitQueryParams>()
+  const [projectOptions] = useState<JiraProjectSelectOption[]>([])
+
+  const {
+    loading,
+    refetch: refetchAcct,
+    error
+  } = useGetAccountNG({ accountIdentifier: accountId, queryParams: { accountIdentifier: accountId } })
```
Comment (chris-harness) [inline src/modules/10-common/pages/Tickets/Tickets.tsx:45]: Doesn't seem like this is needed

File: src/modules/10-common/pages/Tickets/Tickets.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+} from '@common/interfaces/RouteInterfaces'
+// eslint-disable-next-line no-restricted-imports
+import { useVariablesExpression } from '@pipeline/components/PipelineStudio/PiplineHooks/useVariablesExpression'
+import { useGetAccountNG } from 'services/cd-ng'
+// eslint-disable-next-line no-restricted-imports
+import type { JiraProjectSelectOption } from '@pipeline/components/PipelineSteps/Steps/JiraApproval/types'
+// eslint-disable-next-line no-restricted-imports
+import { FormMultiTypeConnectorField } from '@connectors/components/ConnectorReferenceField/FormMultiTypeConnectorField'
+import { FormMultiTypeCheckboxField } from '@common/components'
+// eslint-disable-next-line no-restricted-imports
```
Comment (chris-harness) [inline src/modules/10-common/pages/Tickets/Tickets.tsx:29]: We shouldn't be ignoring restricted imports. Move this into a different module so they become unnecessary.

File: src/modules/10-common/RouteDefinitions.ts [hunk 6]
```diff
@@ -150,6 +150,7 @@ const routes = {
   toAuthenticationSettings: withAccountId(() => '/settings/authentication'),
   toAccountConfiguration: withAccountId(() => '/settings/authentication/configuration'),
   toAccountActivityLog: withAccountId(() => '/settings/authentication/activity-log'),
+  toTickets: withAccountId(() => `/settings/tickets`),
```
Comment (chris-harness) [inline src/modules/10-common/RouteDefinitions.ts:153]:
```suggestion
  toTicketSettings: withAccountId(() => `/settings/tickets`),
```
is probably clearer

File: src/modules/85-sto/pages/TicketSettings/TicketSettings.tsx [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+            }}
+          />
+        </div>
+      </Layout.Horizontal>
+    </Container>
+  )
+
+  function updateTicketSettings(): void {
+    return
+  }
```
Comment (chris-harness) [inline src/modules/85-sto/pages/TicketSettings/TicketSettings.tsx:119]: Now that the API client for Ticket Service is in, you can call it to save these settings

File: src/modules/10-common/strings/strings.en.yaml [hunk 8]
```diff
@@ -1299,3 +1299,11 @@ svcEnvV2Truncated: 'Harness has made a significant update to the services and en
...
 collapse: Collapse
 expand: Expand
+tickets:
+  selectConnector: '[Select] Ticket Provider Connector'
+  selectProjectName: 'Select Project Name'
+  connector: 'Ticket Provider Connector'
+  defaultProjectName: 'Default Project Name'
+  tickets: 'Tickets'
+  ticketComment: 'Append comment to ticket when Security Issue changes are detected (coming soon)'
+  ticketExemption: 'Create a ticket when an Exemption is created (coming soon)'
```
Comment (chris-harness) [inline src/modules/10-common/strings/strings.en.yaml:1309]: Move these to 85-sto?

**Review by chris-harness — APPROVED** [review]

**Review by chris-harness — APPROVED** [review]

**Review by atmollo-harness — APPROVED** [review]

**Review by tan-nhu — APPROVED** [review]

**Pull request merged**